kernel/src/process.rs

```diff
@@ -1473,16 +1473,65 @@ impl<C: Chip> ProcessType for Process<'_, C> {
             let _ = writer.write_fmt(format_args!("{}", config));
         });
 
-        let sram_start = self.memory.as_ptr() as usize;
-        let flash_start = self.flash.as_ptr() as usize;
-        let flash_init_fn = flash_start + self.header.get_init_function_offset() as usize;
+        /// Return Ok(Flash Address, RAM address) if this app has
+        /// fixed_addresses, Err(()) otherwise.
+        fn check_for_fixed_address(flash: &'static [u8]) -> Result<(u32, u32), ()> {
+            // Need to re-parse the TBF header to find if this app has fixed
```

Comment:
This feels like a lot of work and non-trivial duplicated logic. Wouldn't it be easier to just store whether the app is PIC in the `struct ProcessDebug` ?

Reply:
What would happen if #1830 (or something similar) were merged?

Reply:
Oy.. preaching to the choir, but oh what a nightmare `cfg`s are.

By the same token though, if we're in the space-optimized side of the `cfg`, this whole detail-laden panic handler is gone anyway, right?